tests/agents/gherkin-specs/user_agent.feature

```diff
@@ -0,0 +1,19 @@
+Feature: Agent Transport User agent Header
+
+  Scenario: Default user-agent
+    Given an agent
+    When service name is not set
+    When service version is not set
+    Then the User-Agent header matches regex 'elasticapm-(java|ruby|python|dotnet|nodejs|go|php)/[^ ]*'
```

Comment:
Wouldn't `elasticapm-\w+` be sufficiently specific without us requiring to update the spec every time we add a new agent (which admittedly doesn't happen very often)?

Reply:
That's definitely a great idea, I'll update the spec with that.